python/nano/src/bigdl/nano/tf/keras/inference/optimizer.py

```diff
@@ -430,9 +424,9 @@ def quantize(model: Model,
         :return:            A TensorflowBaseModel for INC. If there is no model found, return None.
         """
         invalidInputError(approach == 'static', "Only 'static' approach is supported now.")
-        invalidInputError(y is not None or isinstance(x, tf.data.Dataset),
-                          "y can be omitted only when x is a Dataset which returns \
-                              tuples of (inputs, targets)")
+        if not isinstance(x, tf.data.Dataset) and y is None:
```

Comment:
may also handle the cases that `x` is a `Dataset` but only return feature no label